packages/compile/src/model/read-equations.spec.ts

```diff
@@ -192,6 +195,120 @@ describe('readEquations', () => {
     ])
   })
 
+  it('should work for equations that are affected by separateAllVarsWithDims', () => {
+    const vars = readInlineModel(
+      `
+      DimA: A1, A2 ~~|
+      DimB: B1, B2 ~~|
+      DimC: C1, C2 ~~|
+      x[DimA] = 0 ~~|
+      y[DimB] = 0 ~~|
+      z[DimA, DimB, DimC] = 0 ~~|
+    `,
+      undefined,
+      {
+        separateAllVarsWithDims: [['_dima', '_dimc'], ['_dimb']]
```

Comment:
In adding this test I noticed that the current implementation has a potential issue.  If you were to put the `['_dimb']` first here, then the implementation would hit that case first and would end up separating `z` over `DimB`.

I suppose that this case may not come up often in practice (and probably EPS will be the only user of this option for some time), but I wanted to flag it.  I guess that you can work around it by changing the order of the list like I did here, or falling back to using `specialSeparationDims` for more precise control.